src/Client/ClientBase.cpp

```diff
@@ -352,6 +352,16 @@ ASTPtr ClientBase::parseQuery(const char *& pos, const char * end, bool allow_mu
         res = parseQueryAndMovePosition(*parser, pos, end, "", allow_multi_statements, max_length, settings.max_parser_depth);
     }
 
+    /// We need to change the auth_data before the query will be formatted
+    if (const auto * create_user_query = res->as<ASTCreateUserQuery>())
+    {
+        if (create_user_query->auth_data->getType() == AuthenticationType::NO_PASSWORD && create_user_query->temporary_password)
+        {
+            AuthenticationType default_password_type = global_context->getAccessControl().getDefaultPasswordType();
+            create_user_query->auth_data = AuthenticationData::makePasswordAuthenticationData(default_password_type, create_user_query->temporary_password.value());
```

Comment:
I added a comment in the beginning of this code fragment:
```/// We need to change the auth_data before the query will be formatted```
Otherwise, the client will see `CREATE USER xxx NOT IDENTIFIED`

Reply:
Well, `processOrdinaryQuery()` reformats query in some cases, we can move the code there along with password-complexity checking. We can also create a new function in `ClientBase` which will change `parsed_query` and reformat if necessary.